module/input/Camera/src/Camera.cpp

```diff
@@ -462,12 +228,408 @@ namespace module::input {
+                camera_name));
+        }
+    }
+
+    void disable_camera(CameraContext& camera) {
+        // Disable all streams on the camera
+        stop_streaming(camera);
+
+        // Release pointers in this order to prevent spurious
+        //       g_object_unref: assertion 'G_IS_OBJECT (object)' failed
+        // errors
+        camera.stream.reset();
+        camera.camera.reset();
+        camera.device.reset();
+    }
+
+    void reset_camera(CameraContext& camera) {
+        // Disable the camera
+        disable_camera(camera);
+
+        try {
+            // Create a new context
+            CameraContext new_context = configure_camera(camera.config, camera.reactor);
+
+            // Manually copy context data
+            camera.name   = new_context.name;
+            camera.fourcc = new_context.fourcc;
+            camera.lens   = new_context.lens;
+            camera.Hpc    = new_context.Hpc;
+            camera.camera = new_context.camera;
+            camera.stream = new_context.stream;
+            camera.device = new_context.device;
+            camera.time   = new_context.time;
+
+            // Update the settings on the camera
+            update_settings(camera);
+
+            // Start streaming on the camera
+            start_streaming(camera);
+        }
+        catch (const std::runtime_error& ex) {
+            camera.reactor.log<NUClear::WARN>(
+                fmt::format("Failed to initiate the {} camera: {}", camera.name, ex.what()));
+        }
+    }
+
+    Camera::Camera(std::unique_ptr<NUClear::Environment> environment) : Reactor(std::move(environment)) {
+
+        on<Configuration>("Cameras").then("Configuration", [this](const Configuration& config) {
+            auto serial_number = config["serial_number"].as<std::string>();
+
+            // The camera already exists update its config and then update its settings
+            auto camera = cameras.find(serial_number);
+            if (camera != cameras.end()) {
+                camera->second.config = config;
+                stop_streaming(camera->second);
+                update_settings(camera->second);
+                start_streaming(camera->second);
+            }
+            // The camera doesn't exist yet, so create a blank context and create a watchdog
+            else {
+                const auto camera_name = std::filesystem::path(config.path).stem().string();
+                cameras.emplace(std::make_pair(serial_number,
+                                               CameraContext{*this,
+                                                             config,
+                                                             camera_name,
+                                                             0,  // fourcc is set later
+                                                             XXH32(camera_name.data(), camera_name.size(), 0x4E55436C),
```

Comment:
Magic number again

Reply:
Really we could just use 0, it is a magic number for the hashing algorithm.
It's just other places we've used this magic number fo xxhash and I suppose it was copied here

Reply:
Can it be a `constexpr` in a file imported from `shared/utility`?